SCAutolib/models/file.py

```diff
@@ -0,0 +1,95 @@
+"""
+General operations with generic config files.
+
+This module serves as an interface and defines basic operations
+on config files:
+    create: create content of config file usually based on template.
+            Note, that some child classes may also update content of config
+            file if it already existed.
+    set:    modify values of config files, add keys or sections
+            if necessary
+    save:   save modified content to config file
+    clean:  remove config file; note that some child classes may also restore
+            original config file if backup exists.
+"""
+from pathlib import Path
+from SCAutolib import logger
+from configparser import ConfigParser
+
+
+class File:
+    """
+    This class defines an interface for generic operations on config files
+
+    create: create content of config file based on template file
+    set:    modify content of config files, add keys or sections if necessary
+    save:   save config file
+    clean:  remove config file
+    """
+    _conf_file = None
+    _template = None
+    _default_parser = None
+
+    def __init__(self, filepath: str, template: str = None):
+        """
+        Init of File
+        :param filepath: Path of config file
+        """
+        self._conf_file = Path(filepath)
+        if template is not None:
+            self._template = Path(template)
+
+    def create(self):
+        """
+        Populate internal parser object with content based on template.
+        """
+        if self._conf_file.exists():
+            logger.warning(f"Create error: {self._conf_file} already exists.")
+            raise FileExistsError(f'{self._conf_file} already exists')
```

Comment:
Do we really want to raise an exception? Imagine if the user would want to work with his own **existing** config file, but this exception would be raised for him and the user would be like :expressionless: